### PR TITLE
add professor2

### PR DESCRIPTION
**Pull request by mkirsano**

### CODE REVIEW

**Comment by perrozzi** [comment]
please test

File: professor2-toolfile.spec [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+%install
+
+mkdir -p %i/etc/scram.d
+cat << \EOF_TOOLFILE >%i/etc/scram.d/professor2.xml
+<tool name="professor2" version="@TOOL_VERSION@">
+<client>
+<environment name="PROFESSOR2_BASE" default="@TOOL_ROOT@"/>
+</client>
+<runtime name="PATH" value="$PROFESSOR2_BASE/bin" type="path"/>
+<runtime name="PYTHONPATH" value="$PROFESSOR2_BASE/lib/python@PYTHONV@/site-packages" type="path"/>
```
Comment (davidlt) [inline professor2-toolfile.spec:23]: You are missing these dependencies listed here: py2-numpy py2-sympy py2-matplotlib root cython yoda eigen

File: cmssw-tool-conf.spec [hunk 2]
```diff
@@ -119,6 +119,7 @@ Requires: fftjet-toolfile
 Requires: lapack-toolfile
 Requires: pyminuit2-toolfile
 Requires: professor-toolfile
+Requires: professor2-toolfile
```
Comment (davidlt) [inline cmssw-tool-conf.spec:122]: Do we need to keep old version of professor?

File: professor2.spec [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+Requires: py2-numpy py2-sympy py2-matplotlib root cython yoda eigen
+%prep
+%setup -n Professor-%{realversion}
+
+%build
+make CPPFLAGS=-I${EIGEN_ROOT}/include
+
+%install
+make install PREFIX=%{i}
```
Comment (davidlt) [inline professor2.spec:13]: Does professor2 and professor provide the same binaries and shared libraries? If yes, then we might have an issue.

File: cmssw-tool-conf.spec (hunk 2 above)
Comment (mkirsano) [reply cmssw-tool-conf.spec:122]: There was a request to keep old version

File: professor2.spec (hunk 3 above)
Comment (mkirsano) [reply professor2.spec:13]: The names of binaries (actually scripts) and shared libraries are different (actually the old professor has no library)

File: cmssw-tool-conf.spec (hunk 2 above)
Comment (davidlange6) [reply cmssw-tool-conf.spec:122]: @mkirsano -please summarize the request here (and its expected longevity) - its certainly not normal to try to manage multiple versions of the "same" tool around.

Comment (mkirsano) [reply cmssw-tool-conf.spec:122]: professor2 is very different from professor. It does not yet have all functionality of professor. This is the reason to keep the old version. Note that CMSSW code is not compiled against this external.

**Comment by smuzaffar** [comment]
testing

**Comment by mkirsano** [comment]
still testing?

**Comment by smuzaffar** [comment]
please test
@davidlange6 , this looks good to co-exist with professor.

**Comment by smuzaffar** [comment]
@mkirsano, I see that there is include directory and lib/libProfessor2.so ? Do you want to make these part of professor2-toolfile?

**Comment by mkirsano** [comment]
This is not needed right now (however it would not do any harm).

**Comment by mkirsano** [comment]
Is this still waiting for some other upgrades (python for example)?
Is there automatically the corresponding PR in 90X or it is to be made?

**Pull request closed without merging**